### ISSUE TITLE
JDK 9 - IncompatibleClassChangeError in DockerExtension

### ISSUE THREAD

**Issue by eskatos** [palantir/gradle-docker#96]
Trying to apply the plugin using JDK-9-ea (build 9-ea+144) I get the following exception:

```
java.lang.IncompatibleClassChangeError: Method java.util.Set.of()Ljava/util/Set; must be InterfaceMethodref constant
        at java_util_Set$of.call(Unknown Source)
        at com.palantir.gradle.docker.DockerExtension.<init>(DockerExtension.groovy:33)
        at com.palantir.gradle.docker.DockerExtension_Decorated.<init>(Unknown Source)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:455)
        at org.gradle.internal.reflect.DirectInstantiator.newInstance(DirectInstantiator.java:42)
        ... 93 more
```

Gradle 3.2.1
gradle-docker 0.9.2

**Comment by hogmoru** [comment]
This is a [Groovy bug](https://issues.apache.org/jira/browse/GROOVY-7879): Groovy fails when invoking static methods in interfaces (now allowed by Java 9). I just faced it on my own code, was looking for info and found this issue. The bug is supposed to be fixed, they say fix version is 2.4.11 but I think that's an error (that's the version I'm running and it's still there) and we might have to wait another release.

**Comment by pkoenig10** [comment]
Java 9 issues should be resolved by #157, which upgraded gradle to 4.4 with groovy 2.4.12.
